From 0bc3f97eae3adc759c6604e478ca380d9d2cbfdd Mon Sep 17 00:00:00 2001
From: Damon Liu <damon.liu@mediatek.com>
Date: Wed, 18 Nov 2020 11:06:57 +0800
Subject: [PATCH 2/2] replace-strlcpy-with-strncpy

---
 libsync/sync.c | 28 ++++++++++++++++++----------
 1 file changed, 18 insertions(+), 10 deletions(-)

diff --git a/libsync/sync.c b/libsync/sync.c
index 6b187faed4..8868cfed9d 100644
--- a/libsync/sync.c
+++ b/libsync/sync.c
@@ -134,7 +134,8 @@ static int legacy_sync_merge(const char *name, int fd1, int fd2)
     int ret;

     data.fd2 = fd2;
-    strlcpy(data.name, name, sizeof(data.name));
+    strncpy(data.name, name, sizeof(data.name));
+    data.name[31] = '\0';
     ret = ioctl(fd1, SYNC_IOC_LEGACY_MERGE, &data);
     if (ret < 0)
         return ret;
@@ -147,7 +148,8 @@ static int modern_sync_merge(const char *name, int fd1, int fd2)
     int ret;

     data.fd2 = fd2;
-    strlcpy(data.name, name, sizeof(data.name));
+    strncpy(data.name, name, sizeof(data.name));
+    data.name[31] = '\0';
     data.flags = 0;
     data.pad = 0;

@@ -243,16 +245,19 @@ static struct sync_fence_info_data *sync_file_info_to_legacy_fence_info(
         return NULL;
     legacy_info->len = sizeof(*legacy_info) +
                         num_fences * sizeof(struct sync_pt_info);
-    strlcpy(legacy_info->name, info->name, sizeof(legacy_info->name));
+    strncpy(legacy_info->name, info->name, sizeof(legacy_info->name));
+    legacy_info->name[31] = '\0';
     legacy_info->status = info->status;

     legacy_pt_info = (struct sync_pt_info *)legacy_info->pt_info;
     for (uint32_t i = 0; i < num_fences; i++) {
         legacy_pt_info[i].len = sizeof(*legacy_pt_info);
-        strlcpy(legacy_pt_info[i].obj_name, fence_info[i].obj_name,
+        strncpy(legacy_pt_info[i].obj_name, fence_info[i].obj_name,
                 sizeof(legacy_pt_info->obj_name));
-        strlcpy(legacy_pt_info[i].driver_name, fence_info[i].driver_name,
+        legacy_pt_info[i].obj_name[31] = '\0';
+        strncpy(legacy_pt_info[i].driver_name, fence_info[i].driver_name,
                 sizeof(legacy_pt_info->driver_name));
+        legacy_pt_info[i].driver_name[31] = '\0';
         legacy_pt_info[i].status = fence_info[i].status;
         legacy_pt_info[i].timestamp_ns = fence_info[i].timestamp_ns;
     }
@@ -281,16 +286,19 @@ static struct sync_file_info* legacy_fence_info_to_sync_file_info(
     }
     info->sync_fence_info = (__u64)(uintptr_t)(info + 1);

-    strlcpy(info->name, legacy_info->name, sizeof(info->name));
+    strncpy(info->name, legacy_info->name, sizeof(info->name));
+    info->name[31] = '\0';
     info->status = legacy_info->status;
     info->num_fences = num_fences;

     pt = NULL;
     fence = sync_get_fence_info(info);
     while ((pt = sync_pt_info(legacy_info, pt)) != NULL) {
-        strlcpy(fence->obj_name, pt->obj_name, sizeof(fence->obj_name));
-        strlcpy(fence->driver_name, pt->driver_name,
+        strncpy(fence->obj_name, pt->obj_name, sizeof(fence->obj_name));
+        fence->obj_name[31] = '\0';
+        strncpy(fence->driver_name, pt->driver_name,
                 sizeof(fence->driver_name));
+        fence->driver_name[31] = '\0';
         fence->status = pt->status;
         fence->timestamp_ns = pt->timestamp_ns;
         fence++;
@@ -410,8 +418,8 @@ int sw_sync_fence_create(int fd, const char *name, unsigned value)
     int err;

     data.value = value;
-    strlcpy(data.name, name, sizeof(data.name));
-
+    strncpy(data.name, name, sizeof(data.name));
+    data.name[31] = '\0';
     err = ioctl(fd, SW_SYNC_IOC_CREATE_FENCE, &data);
     if (err < 0)
         return err;
--
2.18.0

